### PR TITLE
 docs(storage): add samples 

### PR DESCRIPTION
**Pull request by HemangChothani**
Add samples:

1.  storage_bucket_delete_default_kms_key

2.   storage_compose_file
3.   storage_create_bucket_class_location
4.   storage_define_bucket_website_configuration
5.   storage_download_public_file
6.   storage_get_service_account
7.   storage_object_csek_to_cmek
8.   storage_object_get_kms_key
9.   storage_set_bucket_public_iam

### CODE REVIEW

File: storage/cloud-client/snippets_test.py [hunk 1]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
     out, _ = capsys.readouterr()
     assert "[]" in out
     assert len(list(bucket.lifecycle_rules)) == 0
+
+
+def test_create_bucket_class_location():
+    bucket = storage_create_bucket_class_location.create_bucket_class_location(
+        "storage-snippets-test"
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:301]: Please use a UUID for bucket names to ensure they don't conflict when running multiple times

File: storage/cloud-client/snippets_test.py [hunk 2]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
...
+    )
+
+    assert bucket.location == 'US'
+    assert bucket.storage_class == "COLDLINE"
+    bucket.delete(force=True)
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:306]: Please use a fixture to delete these bucket, so it's always done even if the test fails or encounters another error. You can use a generator to return a UUID above clean up after the bucket it created.

File: storage/cloud-client/snippets_test.py [hunk 3]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
...
+
+
+def test_bucket_delete_default_kms_key(test_bucket, capsys):
+    storage_bucket_delete_default_kms_key.bucket_delete_default_kms_key(
+        test_bucket.name
+    )
+
+    out, _ = capsys.readouterr()
+    assert test_bucket.default_kms_key_name is None
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:315]: Doesn't look like this tests anything
```suggestion
```

File: storage/cloud-client/snippets_test.py [hunk 4]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
...
+    assert test_bucket.name in out
+
+
+def test_get_service_account(capsys):
+    storage_get_service_account.get_service_account()
+    out, _ = capsys.readouterr()
+
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:322]: nit:
```suggestion

    out, _ = capsys.readouterr()
```

File: storage/cloud-client/snippets_test.py [hunk 5]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
...
+
+def test_download_public_file(test_blob):
+    storage_make_public.make_blob_public(test_blob.bucket.name, test_blob.name)
+    with tempfile.NamedTemporaryFile() as dest_file:
+        storage_download_public_file.download_public_file(
+            test_blob.bucket.name, test_blob.name, dest_file.name
+        )
+
+        assert dest_file.read()
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:333]: Does this actually verify this sample works?  Won't this just ready an empty file if not? 

The preferred pattern for samples is [Arrange, Act, Assert](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#arrange-act-assert):

1. Arrange - prepare the request
2. Act - send request, and receive response
3. Assert - Print response (to show the user how to interact with it) and assert the response occurred when the sample was run

It's an explict non-goal for our samples tests to test the service or product. Instead they should be that the _sample_ ran successfully, and should verify the output of the sample in a reasonable way. Thus, most sample tests should be in the pattern:

1. Run the sample
2. Check the output of the sample

File: storage/cloud-client/snippets_test.py [hunk 6]
```diff
@@ -287,3 +294,81 @@ def test_bucket_lifecycle_management(test_bucket, capsys):
...
+
+
+def test_define_bucket_website_configuration(test_bucket):
+    bucket = storage_define_bucket_website_configuration.\
+        define_bucket_website_configuration(test_bucket.name, "index.html", "404.html")
```
Comment (kurtisvg) [inline storage/cloud-client/snippets_test.py:338]: nit: You can also use `black` to auto format these files
```suggestion
    bucket = storage_define_bucket_website_configuration.define_bucket_website_configuration(
         test_bucket.name, "index.html", "404.html")
```

File: storage/cloud-client/storage_set_bucket_public_iam.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google Inc. All Rights Reserved.
```
Comment (kurtisvg) [inline storage/cloud-client/storage_set_bucket_public_iam.py:3]:
```suggestion
# Copyright 2020 Google LLC. All Rights Reserved.
```

Please adjust everywhere.

File: storage/cloud-client/snippets_test.py (hunk 4 above)
Comment (HemangChothani) [reply storage/cloud-client/snippets_test.py:334]: Done!

**Comment by HemangChothani** [comment]
@frankyn  Could you please review this PR, I am unable to add you as a reviewer.

**Comment by frankyn** [comment]
@JesseLovelace is on point for these reviews. Assigning it to them!

**Comment by busunkim96** [comment]
@HemangChothani Could you investigate the failing test? Do we need to add a new resource to the project?

```
================================== FAILURES ===================================
___________________________ test_object_csek_to_cmek ___________________________
Traceback (most recent call last):
  File "/tmpfs/src/github/python-docs-samples/storage/cloud-client/encryption_test.py", line 122, in test_object_csek_to_cmek
    BUCKET, test_blob_name, TEST_ENCRYPTION_KEY, KMS_KEY
  File "/tmpfs/src/github/python-docs-samples/storage/cloud-client/storage_object_csek_to_cmek.py", line 40, in object_csek_to_cmek
    token, rewritten, total = dest.rewrite(source_blob)
  File "/tmpfs/src/github/python-docs-samples/storage/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/google/cloud/storage/blob.py", line 2098, in rewrite
    timeout=timeout,
  File "/tmpfs/src/github/python-docs-samples/storage/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/google/cloud/_http.py", line 423, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.BadRequest: 400 POST https://storage.googleapis.com/storage/v1/b/python-docs-samples-tests/o/test_blob_2eed0931e61a435a98f112685f291785/rewriteTo/b/python-docs-samples-tests/o/test_blob_2eed0931e61a435a98f112685f291785?destinationKmsKeyName=projects%2Fpython-docs-samples-tests%2Flocations%2Fus%2FkeyRings%2Fgcs-kms-key-ring%2FcryptoKeys%2Fgcs-kms-key: The provided encryption key is incorrect.
--------------------------- Captured stdout teardown ---------------------------
Ignoring 404, detail: 404 DELETE https://storage.googleapis.com/storage/v1/b/python-docs-samples-tests/o/test_blob_2eed0931e61a435a98f112685f291785?generation=1589490963804584: No such object: python-docs-samples-tests/test_blob_2eed0931e61a435a98f112685f291785
```

**Pull request merged**